### PR TITLE
Clean up environment-specific instructions in SETUP.md

### PR DESCRIPTION
**Pull request by Hamms**
Specifically:

- The "general" instructions section had several M1-mac-specific instructions in it; move them into the mac-specific section
- Add instructions for an Ubuntu 22.04-specific issue

Also, we were manually maintaining the numbers of the steps in code, which makes refactoring like this more tedious. By instead using a consistent notation, we can let Markdown handle the numbering for us. Because the numbering change is somewhat noisy, I recommend reviewing this PR commit-by-commit:

- https://github.com/code-dot-org/code-dot-org/pull/46750/commits/df60816758562bd111ca7138f0ca9caa4f6d82a5
- https://github.com/code-dot-org/code-dot-org/pull/46750/commits/dca352f9b1f9d6341c0af6a6adba3fba78b33422

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: SETUP.md [hunk 1]
```diff
@@ -17,32 +17,18 @@ You can do Code.org development using OSX, Ubuntu, or Windows (running Ubuntu in
...
-
-5. `gem install bundler -v 1.17.3`
-
-6. `export LIBRARY_PATH=$LIBRARY_PATH:/opt/homebrew/opt/openssl/lib/`
-
-7. `rbenv rehash`
+1. `rbenv rehash`
 
-8. `bundle install`
+1. `bundle install`
```
Comment (cat5inthecradle) [inline SETUP.md:28]: pretty sure we can't run `bundle install` on M1 without the lockfile changes.

However, I've got another PR that will hopefully remove this workaround.

Comment (Hamms) [reply SETUP.md:28]: Right, but the directions for the lockfile changes are in the section referenced by this earlier step:

    1. Install OS-specific prerequisites
       - See the appropriate section below: [OS X](#os-x-monterey---including-apple-silicon-m1), [Ubuntu](#ubuntu-1804-download-iso), [Windows](#windows)

This section of the instructions is for general instructions which apply to all operating system environments; OS-specific instructions should go in the OS-specific sections

Comment (cat5inthecradle) [reply SETUP.md:28]: oh, i missed one of the 'expand' sections in the diff and it was hiding the M1 heading 🤦

**Review by cat5inthecradle — APPROVED** [review]

File: SETUP.md (hunk 1 above)
Comment (Hamms) [reply SETUP.md:28]: Oh, I should have also mentioned:

Some of the instructions I moved, but some of them I went to move and realized they already existed both in the place I was moving them from and the place I was moving them to, so I just removed them.

**Pull request merged**